src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -3877,7 +3894,7 @@ void processResource(IOperation resource, ArrayBuilder<(IVariableDeclarationOper
             }
         }
 
-        private void AddDisposingFinally(IOperation resource, bool knownToImplementIDisposable, ITypeSymbol iDisposable, bool isAsynchronous)
```

Comment:
>knownToImplementIDisposable [](start = 67, length = 27)

I am not comfortable with elimination of this parameter. There is enough complexity in how it was calculated across both languages that I would prefer us to not take any chances.

Reply:
I can restore it. I originally removed it because the presence of DisposeMethod should be enough to infer it, but I see no harm in being explicit.

Reply:
The other issue was the fact that it doesn't really do what it says anymore. I think we should keep it but re-name it to `requiresRuntimeCheck` because that's all it was really being used for (in the getEnumerator case)

Reply:
>  I originally removed it because the presence of DisposeMethod should be enough to infer it.

Perhaps, that is the case. but how can we be sure that we re-inferred it with 100% accuracy?

---
In reply to: [528044035](https://github.com/dotnet/roslyn/pull/49481#discussion_r528044035) [](ancestors = 528044035)

Reply:
> The other issue was the fact that it doesn't really do what it says anymore. I think we should keep it but re-name it to requiresRuntimeCheck because that's all it was really being used for (in the getEnumerator case)

Renaming it would be fine, of course.

---
In reply to: [528044325](https://github.com/dotnet/roslyn/pull/49481#discussion_r528044325) [](ancestors = 528044325)

Reply:
I am not saying it is impossible to do. I am simply having a hard time proving to myself that we did it without causing a behavior change for some non-obvious scenario.

---
In reply to: [528044366](https://github.com/dotnet/roslyn/pull/49481#discussion_r528044366) [](ancestors = 528044366,528044035)

Reply:
Absolutely. I restored the `knownToImplementIDisposable` and added hold of `IsPattern` too. Then in the builder we have a simple `RuntimeConversionRequired`

Much clearer what's going on and less opportunity to get it wrong.